cdap-ui/app/cdap/components/DataPrep/Directives/MapToTarget/index.js

```diff
@@ -14,36 +14,367 @@
  * the License.
  */
 
-import React, { Component } from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import T from 'i18n-react';
+import { makeStyles } from '@material-ui/core';
+import Input from '@material-ui/core/Input';
+import LinearProgress from '@material-ui/core/LinearProgress';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import { UncontrolledTooltip } from 'reactstrap';
 import classnames from 'classnames';
+import { preventPropagation, connectWithStore } from 'services/helpers';
+import { setPopoverOffset } from 'components/DataPrep/helper';
+import DataPrepStore from 'components/DataPrep/store';
+import {
+  execute,
+  setError,
+  loadTargetDataModelStates,
+  saveTargetDataModelStates,
+  setTargetDataModel,
+  setTargetModel
+} from 'components/DataPrep/store/DataPrepActionCreator';
+
+const useStyles = makeStyles(theme => ({
+  secondLevelPopover: {
+    width: '300px !important',
+  },
+  selectedItem: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  selectedItemLabel: {
+    whiteSpace: 'nowrap',
+    textOverflow: 'ellipsis',
+    overflow: 'hidden',
+    fontWeight: 'bold',
+  },
+  selectedItemName: {
+    flex: 1,
+    whiteSpace: 'nowrap',
+    textOverflow: 'ellipsis',
+    overflow: 'hidden',
+  },
+  unselectIcon: {
+    cursor: 'pointer',
+    padding: theme.spacing(0.75),
+    margin: '0 !important',
```

Comment:
Any specific reason to not use existing styles? Why the override?

Reply:
After refactoring to TypeScript this style was removed.